src/overlays/actors/ovl_Dm_Bal/z_dm_bal.c

```diff
@@ -30,31 +36,178 @@ const ActorInit Dm_Bal_InitVars = {
+    { &object_bal_Anim_00D530, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },
+    { &object_bal_Anim_000C78, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },
+    { &object_bal_Anim_00CB78, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },
+    { &object_bal_Anim_001804, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/DmBal_Draw.s")
+void DmBal_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DmBal* this = THIS;
+
+    this->actor.targetMode = 1;
+    this->actor.uncullZoneForward = 3000.0f;
+    Actor_SetScale(&this->actor, 0.02f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_bal_Skel_00A6D0, &object_bal_Anim_0005FC, this->morphTable,
+                       this->joinTable, OBJECT_BAL_LIMB_MAX);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->unk_334 = 0x3C;
+    this->unk_198 = 0;
+    this->unk_336 = 0;
+    func_80C1EAC4(this);
+}
+
+void DmBal_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_80C1EAC4(DmBal* this) {
+    this->actionFunc = func_80C1EAD8;
+}
+
+void func_80C1EAD8(DmBal* this, GlobalContext* globalCtx) {
+}
+
+void func_80C1EAE8(DmBal* this, GlobalContext* globalCtx) {
+    static u16 D_80C1F2C0 = 0x63;
+    s32 actionIndex;
+
+    if (Cutscene_CheckActorAction(globalCtx, 0x238)) {
+        actionIndex = Cutscene_GetActorActionIndex(globalCtx, 0x238);
+
+        if (D_80C1F2C0 != globalCtx->csCtx.actorActions[actionIndex]->action) {
+            D_80C1F2C0 = globalCtx->csCtx.actorActions[actionIndex]->action;
+            switch (globalCtx->csCtx.actorActions[actionIndex]->action) {
+                case 1:
+                    this->unk_336 = 0;
+                    this->unk_198 = 0;
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, D_80C1F170, 0);
+                    break;
+                case 2:
+                    this->unk_336 = 1;
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, D_80C1F170, 12);
+                    break;
+                case 3:
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, D_80C1F170, 13);
+                    break;
+            }
+        } else if (D_80C1F2C0 == 3) {
+            if (Animation_OnFrame(&this->skelAnime, 0.0f)) {
+                this->unk_336 = 1;
+            } else if (Animation_OnFrame(&this->skelAnime, 29.0f)) {
+                this->unk_336 = 0;
+                this->unk_198 = 0;
+            }
+        }
+        Cutscene_ActorTranslateAndYaw(&this->actor, globalCtx, actionIndex);
+        this->actor.home.pos = this->actor.world.pos;
+    } else {
+        this->unk_336 = 0;
+        this->unk_198 = 0;
+        D_80C1F2C0 = 0x63;
+    }
+}
+
+Vec3f D_80C1F2C4 = { 0.0f, 9.0f, 0.0f };
+u32 D_80C1F2D0[] = { &object_bal_Tex_006050, &object_bal_Tex_0094D0 };
```

Comment:
```suggestion
TexturePtr D_80C1F2D0[] = { object_bal_Tex_006050, object_bal_Tex_0094D0 };
```
Now that we see that these are textures, we should use the `TexturePtr` type as well removing the `&`s.